src/plots/cartesian/set_convert.js

```diff
@@ -198,51 +198,55 @@ module.exports = function setConvert(ax, fullLayout) {
             if(!len) return _l2p(v, ax._m, ax._b);
 
             var isY = axLetter === 'y';
-            var pos = isY ? -v : v;
+            var pos = v;
 
-            var q = 0;
-            for(var i = 0; i < len; i++) {
+            var flip = isY;
+            if(ax.range[0] > ax.range[1]) flip = !flip;
+            var signAx = flip ? -1 : 1;
+
+            var first = 0;
+            var last = len - 1;
+            var q = first;
+            for(var i = first; i <= last; i += 1) {
                 var nextI = i + 1;
                 var brk = ax._rangebreaks[i];
 
-                var min = isY ? -brk.max : brk.min;
-                var max = isY ? -brk.min : brk.max;
+                var min = brk.min;
+                var max = brk.max;
 
-                if(pos < min) break;
-                if(pos > max) q = nextI;
+                if(signAx * pos < signAx * min) break;
+                if(signAx * pos > signAx * max) q = nextI;
                 else {
                     // when falls into break, pick 'closest' offset
-                    q = pos > (min + max) / 2 ? nextI : i;
+                    q = signAx * pos > signAx * (min + max) / 2 ? nextI : i;
```

Comment:
`l2p` is a hot path, so it's worth optimizing a good deal - even if it means some duplicated code, though I'm not sure if that would help here, and even if the code needs more comments and is less self-documenting. For example:
- `signAx * pos` could be done once up front, and `var min = signAx * brk.min` etc would simplify these conditionals a bit
- `first` and `last` look unnecessary
- Flip `_m2` for y axes after using it to calculate `_B`, so we don't need the `(isY ? -1 : 1) *` down below.
- Move `var isY = axLetter === 'y'` to the outer scope, where `axLetter` was defined.

`p2l` I think is generally not as hot, though maybe for some traces it's used in hover? Not sure. (Note that `c2p` uses `l2p` and `p2c` uses `p2l`)

Reply:
Good calls. Addressed in the commits below.